pkgs/applications/emulators/wine/sources.nix

```diff
@@ -115,12 +115,12 @@ in rec {
 
   wayland = fetchFromGitLab rec {
     # https://gitlab.collabora.com/alf/wine/-/tree/wayland
-    version = "7.20";
-    sha256 = "sha256-UrukAnlfrr6eeVwFSEOWSVSfyMHbMT1o1tfXxow61xY=";
+    version = "8.0";
+    hash = "sha256-xjk3Eq1aatF+0rDK09cEbIQAhSWCm67Sz5JXlIESuUs=";
     domain = "gitlab.collabora.com";
     owner = "alf";
     repo = "wine";
-    rev = "1dc9821ef0b6109c74d0c95cd5418caf7f9feaf1";
+    rev = "6677c044abb711f423fadfb61f2c9a37da6ff686";
```

Comment:
I don't think this commit contains the Wayland driver; it looks like this is just the upstream 8.0 release commit, without the Wayland patches. Did the update script supply this or did you set it manually?

Reply:
I didn't run the update script because of the major version bump; I assumed I had to manually update this variant as well.
How does the script work in this case? It just pulls HEAD, right? How do we know which commit from the range that has `8.0` in the version file to pick? Is it just always updated with the wayland patches on top? I don't use this wine variant at all myself so I normally don't touch it but because of the removal of options I felt I needed to this time.
From the current state the update script moves to rev to 2f80bd757739f2dd8da41abceae6b87d2c568152, is that correct?

Reply:
The update script for wine-wayland, unlike the other update scripts, doesn't pay attention to major version because I couldn't think of a reasonable way for it to do so—the Git history in the Collabora repo is frequently and aggressively rewritten. The Wayland patches do seem to be added on top but also patches seem to be inserted into the middle of that sequence sometimes (I submitted one by email once and I think I had to go back at least several months in the patch history to confirm it landed) so I suspect the team maintaining that fork has a more complex process going on behind the scenes than simply rebasing periodically.

So yeah, the script just pulls HEAD, checks the version in the file, and makes that the new version number for wine-wayland hoping that everything else works out all right. This is not ideal, but wine-wayland is experimental anyway and it's probably good enough most of the time IMO.

2f80bd75... is what I'd expect at the current time, yup.

Reply:
Thanks for clarifying and checking; I've changed the rev/hash and rebased it into the original update commit.